File: bfs/bfs.go
package main

import (
	"fmt"
)

func bfs(graph map[int][]int, startNode int) []int {
	visited := make(map[int]bool)
	queue := []int{startNode}
	result := []int{}

	for len(queue) > 0 {
		node := queue[0]
		queue = queue[1:]

		if !visited[node] {
			visited[node] = true
			result = append(result, node)
			queue = append(queue, graph[node]...)
		}
	}

	return result
}

func main() {
	graph := map[int][]int{
		0: {1, 2},
		1: {0, 3, 4},
		2: {0, 5},
		3: {1},
		4: {1},
		5: {2},
		6: {7},
		7: {6, 8},
		8: {7, 9},
		9: {8},
	}

	fmt.Println(bfs(graph, 0))
}

File: bfs/bfs.java
import java.util.*;

public class bfs {
    public static List<Integer> bfs(Map<Integer, List<Integer>> graph, int startNode) {
        Set<Integer> visited = new HashSet<>();
        Queue<Integer> queue = new LinkedList<>();
        List<Integer> result = new ArrayList<>();
        
        queue.add(startNode);
        
        while (!queue.isEmpty()) {
            int node = queue.poll();
            if (!visited.contains(node)) {
                visited.add(node);
                result.add(node);
                queue.addAll(graph.getOrDefault(node, new ArrayList<>()));
            }
        }
        
        return result;
    }

    public static void main(String[] args) {
        Map<Integer, List<Integer>> graph = new HashMap<>();
        graph.put(0, Arrays.asList(1, 2));
        graph.put(1, Arrays.asList(0, 3, 4));
        graph.put(2, Arrays.asList(0, 5));
        graph.put(3, Arrays.asList(1));
        graph.put(4, Arrays.asList(1));
        graph.put(5, Arrays.asList(2));
        graph.put(6, Arrays.asList(7));
        graph.put(7, Arrays.asList(6, 8));
        graph.put(8, Arrays.asList(7, 9));
        graph.put(9, Arrays.asList(8));
        
        System.out.println(bfs(graph, 0));
    }
}
